requirements_test.txt

```diff
@@ -1,11 +1,10 @@
-# Note: using < here to maintain compatability with Python3.6
-# See: https://github.com/pyvista/pyvista/pull/2038
+# Note: using < here to reduce breakages
 
 Sphinx<4.5.0
 cmocean<2.1
 codecov<2.2.0
 colorcet<3.1.0
-hypothesis<6.31.6  # not latest due to python3.6 compatibility
+hypothesis<6.31.6
```

Comment:
So is this the latest one or is it not? Especially with dependabot I'm unsure. It's possible that this comment was stale.

Reply:
6.37.2 is the latest. It might be worth trying to bump all of these requirements in a separate PR, though. Then it's clear additional changes are due to version bumps and not directly related to dropping 3.6 support. I would probably try it as a separate PR where *all* dependencies were bumped to their latest release version and see if things worked...

It might even be worth having a separate CI with all latest versions of dependencies (maybe marked as an allowed failure?) to stay on top of errors, if you don't already have that. We find breaking changes in upstream libraries frequently (once or twice a month?) in MNE-Python that way. Sometimes it's something we need to fix, sometimes it is a bad upstream change.

Reply:
Ahh okay, didn't realize that `dependabot` took care of this stuff. I have no idea how it works -- since `hypothesis` was removed from the list of dependabot exceptions, should it have complained about this line not being up to date...?

Reply:
> I'd like to do that in a separate PR as soon as we merge this one.

Agreed a second PR is the way to go for this one. FYI I'd recommend using the nightly builds of NumPy and matplotlib -- some of these lines could be reused:

https://github.com/mne-tools/mne-python/blob/4396a3f7f073ba735d7e578f449496676fdc2bd7/tools/github_actions_dependencies.sh#L22-L31

Reply:
@larsoner dependabot automatically opens PRs that bump dependency versions when new releases are available. I assume it would only realise the hypothesis update when this is merged, removing the exception.

Reply:
Oh, I left out the explanation about this place. I decided that version updates should be left to dependabot. Therefore, I only deleted the comment.